typing/typecore.ml

```diff
@@ -4068,8 +4068,8 @@ and type_expect_
         rt, funct
       in
       let type_sfunct_args sfunct extra_args =
-        match sfunct.pexp_desc with
-        | Pexp_apply (sfunct, args) ->
+        match Extensions.Expression.of_ast sfunct, sfunct.pexp_desc with
+        | None, Pexp_apply (sfunct, args) ->
```

Comment:
This is a good idea for minimal diff noise when you want to share a default case!  I like it.  (Mark as resolved when read.)